### PR TITLE
Sameera/search

### PR DESCRIPTION
**Pull request by sameerav**

### CODE REVIEW

File: app/controllers/api/searchables_controller.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    render json: people_array
+  end
+
+  def roster
+    students = Student.all
+    teachers = Teacher.all
+    people = (students + teachers).sort_by &:first_name
+    render json: people
+  end
+end
```
Comment (sonia-y) [inline app/controllers/api/searchables_controller.rb]: new line

File: app/assets/javascripts/components/admin/roster_page.es6.jsx [hunk 2]
```diff
@@ -16,20 +32,42 @@ class RosterPage extends React.Component {
...
-        <AdminHeader />
-        <div className="content-wrapper roster-page">
-          <h1 className="roster-title">Roster</h1>
-          <div className="roster-container">
-            {this.renderPeople()}
+  searchOnChange(input) {
+    if (input === "") {
+      this.loadRoster();
+    } else {
+      var route = ApiConstants.searchables.users(input);
```
Comment (sonia-y) [inline app/assets/javascripts/components/admin/roster_page.es6.jsx]: let's get in the habit of using const, apply this to route, resolve, reject

File: app/assets/javascripts/components/admin/roster_page.es6.jsx [hunk 3]
```diff
@@ -16,20 +32,42 @@ class RosterPage extends React.Component {
...
 
-  render () {
-    return (
-      <div className="page-wrapper">
-        <AdminHeader />
-        <div className="content-wrapper roster-page">
-          <h1 className="roster-title">Roster</h1>
-          <div className="roster-container">
-            {this.renderPeople()}
+  searchOnChange(input) {
```
Comment (sonia-y) [inline app/assets/javascripts/components/admin/roster_page.es6.jsx]: onSearchChange

File: app/assets/javascripts/components/admin/roster_page.es6.jsx [hunk 4]
```diff
@@ -16,20 +32,42 @@ class RosterPage extends React.Component {
...
+  render() {
+    if (this.state.people == null) {
+      return (
+        <div />
+      );
+    } else {
+      return (
+        <div className="page-wrapper">
+          <AdminHeader />
+          <SearchBar onChange = {(input) => this.searchOnChange(input)} />
```
Comment (sonia-y) [inline app/assets/javascripts/components/admin/roster_page.es6.jsx]: I'm not really sure if you need to create a separate component for the search bar. I think you can just use a FormControl (check student_form file) and keep track of the input as a state in this component

File: app/controllers/api/searchables_controller.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+      if result.searchable_type == "Teacher"
+        people_array.push(Teacher.find(result.searchable_id))
+      else
+        people_array.push(Student.find(result.searchable_id))
+      end
+    end
+    render json: people_array
+  end
+
+  def roster
```
Comment (sonia-y) [inline app/controllers/api/searchables_controller.rb]: Why is roster in searchables? shouldn't it be in admin controller? Looks like you're not doing anything related to searching in this function

File: app/assets/javascripts/components/admin/roster_page.es6.jsx [hunk 6]
```diff
@@ -1,11 +1,27 @@
...
+    this.state = {
+      people: null,
     };
   }
 
+  componentDidMount() {
+    this.loadRoster();
+  }
+
+  loadRoster() {
```
Comment (sameerav) [inline app/assets/javascripts/components/admin/roster_page.es6.jsx]: ok oldsonia

File: app/controllers/api/searchables_controller.rb (hunk 5 above)
Comment (sameerav) [inline app/controllers/api/searchables_controller.rb]: because its an api thing

File: app/controllers/api/searchables_controller.rb (hunk 1 above)
Comment (sameerav) [inline app/controllers/api/searchables_controller.rb]: NO

**Pull request merged**